sklearn/preprocessing/imputation.py

```diff
@@ -375,3 +375,203 @@ def transform(self, X):
             X[coordinates] = values
 
         return X
+
+
+class MissingIndicator(BaseEstimator, TransformerMixin):
+    """Missing values indicator transformer for indicating missing values.
+
+    Parameters
+    ----------
+    missing_values : integer or "NaN", optional (default="NaN")
+        The placeholder for the missing values. All occurrences of
+        `missing_values` will be imputed. For missing values encoded as np.nan,
+        use the string value "NaN".
+
+    features : {'train' (default), 'all', array-like of int}
+        If "all", mask will represent all features.
+        If "train", mask will only represent features with missing values
+        during fit time.
+        If mask/indices, mask will only represent features in the
+        indices or mask.
+
+    sparse : boolean or "auto", optional (default="auto")
+        If True, the transformed ``X`` will be a sparse matrix.
+        If False, the transformed ``X`` will be a numpy array.
+        If "auto", the transformed ``X`` will be of same type as input.
+
+    Example
+    -------
+    >>> from sklearn.preprocessing import MissingIndicator
+    >>> import numpy as np
+    >>> X1 = np.array([
+    ...       [-1,  1,  3],
+    ...       [ 4,  0, -1],
+    ...       [ 8,  1,  0]
+    ... ])
+    >>> X2 = np.array([
+    ...       [ 5, -1, -1],
+    ...       [-1,  2,  3],
+    ...       [ 2,  4,  0]
+    ... ])
+    >>> indicator = MissingIndicator(missing_values=-1)
```

Comment:
In this example, the transformer is fit on X1 while X2 is being transformed. And regarding NaN, there is no reason to use -1. Sorry will change it too.